cairo_vm_hints/src/hints/lib/verifiers/evm/account_verifier.rs

```diff
@@ -0,0 +1,230 @@
+    let batch = exec_scopes.get::<Proofs>("batch")?;
+    let idx: usize = get_integer_from_var_name(vars::ids::IDX, vm, &hint_data.ids_data, &hint_data.ap_tracking)?
+        .try_into()
+        .unwrap();
+    let account = batch.accounts[idx].clone();
+    let account_key = BigUint::from_str_radix(&account.account_key.to_string(), 16).unwrap();
+
+    let (key_low, key_high) = split_128(&account_key);
+    insert_value_from_var_name(
+        vars::ids::KEY_LOW,
+        MaybeRelocatable::Int(Felt252::from(key_low)),
+        vm,
+        &hint_data.ids_data,
+        &hint_data.ap_tracking,
+    )?;
+    insert_value_from_var_name(
+        vars::ids::KEY_HIGH,
+        MaybeRelocatable::Int(Felt252::from(key_high)),
+        vm,
+        &hint_data.ids_data,
+        &hint_data.ap_tracking,
+    )?;
+
+    Ok(())
+}
+
+pub const HINT_ACCOUNT_KEY_LEADING_ZEROS: &str =
+    "ids.key_leading_zeros = len(account.account_key.lstrip(\"0x\")) - len(account.account_key.lstrip(\"0x\").lstrip(\"0\"))";
+
+pub fn hint_account_key_leading_zeros(
+    vm: &mut VirtualMachine,
+    exec_scopes: &mut ExecutionScopes,
+    hint_data: &HintProcessorData,
+    _constants: &HashMap<String, Felt252>,
+) -> Result<(), HintError> {
+    let batch = exec_scopes.get::<Proofs>("batch")?;
+    let idx: usize = get_integer_from_var_name(vars::ids::IDX, vm, &hint_data.ids_data, &hint_data.ap_tracking)?
+        .try_into()
+        .unwrap();
+    let account = batch.accounts[idx].clone();
+
+    let key_leading_zeros = count_leading_zero_nibbles_from_hex(&account.account_key.to_string());
+    insert_value_from_var_name(
+        vars::ids::KEY_LEADING_ZEROS,
+        MaybeRelocatable::Int(Felt252::from(key_leading_zeros)),
+        vm,
+        &hint_data.ids_data,
+        &hint_data.ap_tracking,
+    )
+}
+
+pub const HINT_ACCOUNT_PROOFS_LEN: &str = "len(account.proofs)";
+
+pub fn hint_account_proofs_len(
+    vm: &mut VirtualMachine,
+    exec_scopes: &mut ExecutionScopes,
+    hint_data: &HintProcessorData,
+    _constants: &HashMap<String, Felt252>,
+) -> Result<(), HintError> {
+    let batch = exec_scopes.get::<Proofs>("batch")?;
+    let idx: usize = get_integer_from_var_name(vars::ids::IDX, vm, &hint_data.ids_data, &hint_data.ap_tracking)?
+        .try_into()
+        .unwrap();
+    let account = batch.accounts[idx].clone();
+
+    insert_value_into_ap(vm, Felt252::from(account.proofs.len()))
+}
+
+pub const HINT_ACCOUNT_PROOF_AT: &str = "proof = account.proofs[ids.idx]";
+
+pub fn hint_account_proof_at(
+    vm: &mut VirtualMachine,
+    exec_scopes: &mut ExecutionScopes,
+    hint_data: &HintProcessorData,
+    _constants: &HashMap<String, Felt252>,
+) -> Result<(), HintError> {
+    let batch = exec_scopes.get::<Proofs>("batch")?;
+    let idx: usize = get_integer_from_var_name(vars::ids::IDX, vm, &hint_data.ids_data, &hint_data.ap_tracking)?
+        .try_into()
+        .unwrap();
+    let proof = batch.accounts[idx].proofs[idx].clone();
```

Comment:
here idx (account) != idx (proofs) in the context of the cairo0

Reply:
Do you mean that we should retrieve `account` by reference (variable name) after its definition in the following hint:
```py
account = batch.accounts[ids.idx]
```
?